### PR TITLE
Feature/rovecomm parse

### PR DESCRIPTION
**Pull request by Yajirushi12**

### CODE REVIEW

**Comment by Yajirushi12** [comment]
Going to add String decoding in a near future PR, wanted to get the fully functional number decoding into dev asap to support development by others.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (Yajirushi12) [inline Electron/Core/RoveProtocol/Rovecomm.ts]: We probably need to consider what exactly to return as. Number, BigInt, and array of uint/int 8/16/32?

Comment (Yajirushi12) [reply Electron/Core/RoveProtocol/Rovecomm.ts]: I think standardizing on either Number or BigInt is probably the best way to go to provide a sufficiently abstracted interface for other components and not require them to know which data type they're receiving.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (Yajirushi12) [inline Electron/Core/RoveProtocol/Rovecomm.ts]: I don't think using a readUint or readInt function here is the correct course of action, We have some packets which contain signed information versus some that contain unsigned information under last years standards iirc.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (Yajirushi12) [inline Electron/Core/RoveProtocol/Rovecomm.ts:25]: Shouldn't this switch on DataTypes[dataType]?

**Comment by thomas-gleiforst** [comment]
Note: This PR has expanded beyond just "rovecomm parse" and should now be all rovecomm functionality other than outgoing TCP connections.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (thomas-gleiforst) [reply Electron/Core/RoveProtocol/Rovecomm.ts:25]: I don't believe so.. At least not the way we have the rest of the switch set up currently. From Rovecomm Manifest, dataType is the number from the enum, such as 6 representing float. So we should switch on the number 6, and figure out which enum in DataTypes = 6. Correct?

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (EliVerbrugge) [inline Electron/Core/RoveProtocol/Rovecomm.ts:205]: Are you handling a packet that cannot be found in the manifest? Right now it looks like you might continue with incomplete data.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (EliVerbrugge) [inline Electron/Core/RoveProtocol/Rovecomm.ts]: dataLength may be a little ambiguous, it's technically the data count. Length might imply that its the size in bytes, while it's the number of seperate data entries. Not a big deal either way.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (EliVerbrugge) [inline Electron/Core/RoveProtocol/Rovecomm.ts:115]: Do you want to handle telemetry we can't find in the manifest? Or is a null dataIdStr event good enough?

**Comment by Yajirushi12** [comment]
I'm probably going to close this PR without merging soon, I was intending for it to be merged after parsing was completed but it looks like we're going to be putting all of the rovecomm implementation in this branch before merging.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (Yajirushi12) [reply Electron/Core/RoveProtocol/Rovecomm.ts:115]: With the current structure, the emit with a specific string will be used for easy consumption of messages by down-stream components, so leaving it as null will just prevent any component from accessing the data. It will still be logged through the "all" emit just after.

It would probably be best to have an additional warning in the event that a "null" emit occurs to provide real-time feedback as to packet definitions varying between the board implementation and the basestation implementation though.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (Yajirushi12) [reply Electron/Core/RoveProtocol/Rovecomm.ts]: Yeah, we should probably switch this to Data Count in the two current instances to ensure easier readability for future maintainers.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (thomas-gleiforst) [reply Electron/Core/RoveProtocol/Rovecomm.ts:115]: Emitting this null event will log the data and the dataId to the Console. The hope is that this will be enough information for someone at basestation to diagnose this issue and fix the board code or add the dataId to the manifest.

File: Electron/Core/RoveProtocol/Rovecomm.ts
Comment (thomas-gleiforst) [reply Electron/Core/RoveProtocol/Rovecomm.ts:205]: We were not. This has been fixed.

**Review by thomas-gleiforst — APPROVED** [review]

**Comment by thomas-gleiforst** [comment]
> I'm probably going to close this PR without merging soon, I was intending for it to be merged after parsing was completed but it looks like we're going to be putting all of the rovecomm implementation in this branch before merging.

While the name of the PR is no longer accurate, we do still plan to merge these features today. Please leave open.

**Pull request merged**
